### PR TITLE
Fixed #24414 -- Added examples of Prefetch object usage to the docs.

### PR DESCRIPTION
**Pull request by sww**
As always, feedback and comments are welcome.

### CODE REVIEW

**Comment by charettes** [comment]
Hi @sww, thanks for the documentation additions.

I recently helped some users on the mailing list who were trying to figure out how they could correctly rely on prefetch to reduce their number of queries when dealing with m2m relationship and I'm convinced they could have benefited from these extra examples.

If I could suggest a minor addition I would add an example (or alter one you just added) where many objects are fetched (`get()` is not used) and `prefetch_related()` is used. I would explain that in this case **only two queries** will be executed independently of the number of fetched and related objects compared to `N+1` queries if related objects are not prefetched.

**Comment by sww** [comment]
@charettes 

Added an example, does that look good?

File: docs/ref/models/querysets.txt [hunk 1]
```diff
@@ -2862,15 +2862,35 @@ The ``Prefetch()`` object can be used to control the operation of
 
 The ``lookup`` argument describes the relations to follow and works the same
 as the string based lookups passed to
-:meth:`~django.db.models.query.QuerySet.prefetch_related()`.
+:meth:`~django.db.models.query.QuerySet.prefetch_related()`.  For example:
```
Comment (charettes) [inline docs/ref/models/querysets.txt]: One space only before `For example`:

File: docs/ref/models/querysets.txt [hunk 2]
```diff
@@ -2862,15 +2862,35 @@ The ``Prefetch()`` object can be used to control the operation of
...
+
+    >>> Question.objects.prefetch_related(Prefetch('choice_set')).get().choice_set.all()
+    [<Choice: Not much>, <Choice: The sky>, <Choice: Just hacking again>]
+    # This will only execute two queries regardless of the number of results
+    # or related objects.
```
Comment (charettes) [inline docs/ref/models/querysets.txt]: It would say `regardless of the number of Question and Choice objects`.

**Comment by charettes** [comment]
Except those two minor changes the patch LGTM. I guess a this should be reviewed a by a native English speaker before being merged.

**Comment by collinanderson** [comment]
The English seems fine.

**Comment by charettes** [comment]
Merged in a3e89f13dfb1f22a26ead8b06b37695598a4421a, thans!

**Pull request closed without merging**